src/ensembl/src/shared/components/instant-download/instant-download-fetch/fetchForGene.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { wrap } from 'comlink';
+
+import downloadAsFile from 'src/shared/helpers/downloadAsFile';
+
+import { TranscriptOptions } from '../instant-download-transcript/InstantDownloadTranscript';
+import { fetchGeneSequenceMetadata } from './fetchSequenceChecksums';
+
+import { WorkerApi } from 'src/shared/workers/sequenceFetcher.worker';
+import {
+  getGenomicSequenceData,
+  prepareDownloadParameters
+} from './fetchForTranscript';
+import { flattenDeep } from 'lodash';
+
+type GeneOptions = {
+  transcript: Partial<TranscriptOptions>;
+  gene: {
+    genomicSequence: boolean;
+  };
+};
+
+type FetchPayload = {
+  genomeId: string;
+  geneId: string;
+  options: GeneOptions;
+};
+
+export const fetchForGene = async (payload: FetchPayload) => {
+  const {
+    genomeId,
+    geneId,
+    options: { transcript: transcriptOptions, gene: geneOptions }
+  } = payload;
+
+  const geneSequenceData = await fetchGeneSequenceMetadata({
+    genomeId,
+    geneId
+  });
+
+  const sequenceDownloadParams = flattenDeep(
+    geneSequenceData.transcripts.map((transcript) =>
+      prepareDownloadParameters({
+        transcriptSequenceData: transcript,
+        options: transcriptOptions
+      })
+    )
+  );
```

Comment:
I'm confused: what's being flattened here?

Also, if there's something really being flattened, won't the native Array.prototype.flat or .flatMap do the job?

Reply:
I used `flatMap` for this ✅